### PR TITLE
feat [IDP-1612] : Added the docs for IDP Stage

### PR DESCRIPTION
**Pull request by Debanitrkl**
# Harness Developer Pull Request
Added the docs for IDP Stage

## What Type of PR is This?

- [ ] Issue
- [x] Feature
- [ ] Maintenance/Chore

If tied to an Issue, list the Issue(s) here:

* *Issue(s)*

## House Keeping
Some items to keep track of. Screen shots of changes are optional but would help the maintainers review quicker. 

- [x] Tested Locally
- [ ] *Optional* Screen Shoot.

### CODE REVIEW

File: docs/internal-developer-portal/flows/idp-stage.md [hunk 1]
```diff
@@ -0,0 +1,661 @@
...
+
+The self-service flow in IDP is powered by the Harness Pipelines. A stage is a part of a pipeline that contains the logic to perform a major segment of a larger workflow defined in a pipeline. Stages are often based on the different workflow milestones, such as building, approving, and delivering.
+
+The process of adding a stage to a pipeline is the same for all Harness modules. When you add a stage to a pipeline, you select the stage type, such as **Developer Portal** for IDP or **Build** for CI or **Deploy** for CD. The available stage settings are determined by the stage type, and, if applicable, the module associated with the selected stage type.
+
+This functionality is limited to the modules and settings that you have access to.
+
+## How to Add the Developer Portal Stage
+
+1. Change the module to **Continious Integration**, and select the **Pipeline** from side nav and **Create a Pipeline**.
```
Comment (OrkoHunter) [inline docs/internal-developer-portal/flows/idp-stage.md:26]: In the old nav - let's recommend to use the "Projects" tab for pipelines, and not any particular module like CI.
In the new nav - we have pipelines in IDP Admin section.

File: docs/internal-developer-portal/flows/idp-stage.md [hunk 2]
```diff
@@ -0,0 +1,661 @@
...
+
+3. Now **Select Stage Type** as **Developer Portal** and add a [name for your stage](https://developer.harness.io/docs/platform/pipelines/add-a-stage/#stage-names) to **Set Up Stage**.
+
+![](./static/dev-portal-stage-selection.png)
+
+![](./static/set-up-stage.png)
+
+## Infrastructure
+
+4. Under **Infrastructure** tab we suggest you to select **[Harness Cloud](https://developer.harness.io/docs/continuous-integration/use-ci/set-up-build-infrastructure/use-harness-cloud-build-infrastructure#use-harness-cloud)** but you're free to choose from [Kubernetes](https://developer.harness.io/tutorials/ci-pipelines/kubernetes-build-farm#define-the-build-infrastructure) or [Local](https://developer.harness.io/docs/continuous-integration/use-ci/set-up-build-infrastructure/define-a-docker-build-infrastructure#set-the-pipelines-build-infrastructure) or [VM](https://developer.harness.io/docs/category/set-up-vm-build-infrastructures)
```
Comment (OrkoHunter) [inline docs/internal-developer-portal/flows/idp-stage.md:44]: Add a note about free 2000 minutes build credits and docs for it.

File: docs/internal-developer-portal/flows/idp-stage.md [hunk 3]
```diff
@@ -0,0 +1,661 @@
...
+
+![](./static/infrastructure.png)
+
+## Execution Steps
```
Comment (OrkoHunter) [inline docs/internal-developer-portal/flows/idp-stage.md:48]: Good to show the final pipeline here as a screenshot.

**Review by OrkoHunter — COMMENTED** [review]
This feels like a reference page + tutorial. Should we split this as such?

Also, do you think we should update our default tutorial to now use the stage? This is going to be released behind a feature flag.

File: docs/internal-developer-portal/flows/idp-stage.md (hunk 2 above)
Comment (Debanitrkl) [reply docs/internal-developer-portal/flows/idp-stage.md:44]: Will add a note, also the docs links I have added links to the same.

**Review by devesh-harness — APPROVED** [review]

**Pull request merged**